src/System.Runtime.Extensions/src/System/IO/Path.Unix.cs

```diff
@@ -69,15 +147,38 @@ private static string RemoveLongPathPrefix(string path)
 
         public static string GetTempPath()
         {
-            // TODO: Implement this
-            throw new NotImplementedException();
+            const string TempEnvVar = "TMPDIR";
+            const string DefaultTempPath = "/tmp/";
+
+            // Get the temp path from the TMPDIR environment variable.
+            // If it's not set, just return the default path.
+            // If it is, return it, ensuring it ends with a slash.
+            string path = Environment.GetEnvironmentVariable(TempEnvVar);
+            return
+                string.IsNullOrEmpty(path) ? DefaultTempPath :
+                IsDirectorySeparator(path[path.Length - 1]) ? path :
+                path + DirectorySeparatorChar;
         }
 
         private static string InternalGetTempFileName(bool checkHost)
         {
-            // TODO: Implement this
-            throw new NotImplementedException();
-        }s
+            const string Suffix = ".tmp";
+            const int SuffixByteLength = 4;
+
+            // mkstemps takes a char* and overwrites the XXXXXX with six characters
+            // that'll result in a unique file name.
+            string template = GetTempPath() + "tmpXXXXXX" + Suffix + "\0";
+            byte[] name = Encoding.UTF8.GetBytes(template);
```

Comment:
Is there a reason you have to take this approach instead of using StringBuilder?  Does the marshaller not correctly convert things to UTF-8 and back in that case?